converters/toZigbee.js

```diff
@@ -2680,6 +2680,21 @@ const converters = {
             }
         },
     },
+    tuya_thermostat_away_mode: {
+        key: ['away_mode'],
+        convertSet: async (entity, key, value, meta) => {
+            // HA has special behavior for the away mode
+            const awayPresetId = utils.getKeyByValue(utils.getMetaValue(entity, meta.mapped, 'tuyaThermostatPreset'), 'away', null);
+            if (awayPresetId !== null) {
+                if (value == 'ON') {
+                    await sendTuyaCommand(entity, 1028, 0, [1, parseInt(awayPresetId)]);
+                }
+                // In case 'OFF' tuya_thermostat_preset() should be called with another preset
```

Comment:
In the `etop` converter you also added the `OFF` case (https://github.com/Koenkk/zigbee-herdsman-converters/blob/d6b672cdc268a93d2d30196122f40cdd02e23421/converters/toZigbee.js#L2575). If I understand correctly, sending `away_mode: OFF` will have no effect now?

Reply:
In the ETOP converter there is no other presets than away and no preset will be published to the `preset` topic. So, I'm switching to 'manual' mode.
`TS0601` has a lot of other presets, I guess, that after switching OFF away mode new preset should be published to `preset` topic from the HA.
If you don't think so, I can set other preset manually, which one of them: `'schedule', 'manual', 'away', 'boost', 'complex', 'comfort', 'eco'`?

Reply:
I don't know, what will be published first, `OFF` to  `away_mode` or new preset to `preset`

Reply:
Let's set it to `schedule`.